### PR TITLE
Update pages about developing extensions for Firefox for Android to reflect current status with Fenix 

### LINKED ISSUE #735: Update instructions for running extensions on Fenix (Android)
https://extensionworkshop.com/documentation/develop/developing-extensions-for-firefox-for-android/
https://extensionworkshop.com/documentation/develop/getting-started-with-web-ext/#testing-in-firefox-for-android

The content is mostly correct, except for some minor details. We should correct this and remove the notice.
Ideally we should keep the original documentation (somewhere else), in case someone ever needs to refer to it.

### PR DESCRIPTION
**Pull request by caitmuenster**
Fixes #735. 

This PR is intended to update information about extension development for Firefox for Android Fenix to what we currently know. 

- Splits development instructions for Fennec onto its own page 
- Removes information that only applies to Fennec from the `/documentation/develop/developing-extensions-for-firefox-for-android/` page 
- Updates alert messages about extensions in Fenix 

@rpl, can you review the content of the testing instructions and @muffinresearch, can you review overall changes? Thanks!

### CODE REVIEW

File: src/content/documentation/develop/developing-extensions-for-firefox-for-android.md [hunk 1]
```diff
@@ -6,20 +6,22 @@ topic: Develop
...
+At the time of launch, only a limited number of [Recommended Extensions](https://support.mozilla.org/en-US/kb/recommended-extensions-program?utm_source=extensionworkshop.com&utm_medium=dev-article&utm_content=developing-extensions-for-firefox-for-android) will be supported. We would like to support more extensions and we are currently evaluating how to best do that.
 
-This article describes the development process of extensions on Firefox for Android through version 68. These methods are not supported for versions 69+.
+Currently, arbitrary extensions can only be loaded temporarily in Fenix. We are working on supporting a pref on [Firefox for Android Nightly](https://play.google.com/store/apps/details?id=org.mozilla.fenix) to enable persistent loading for arbitrary extensions. You can follow progress by subscribing to [this issue](https://github.com/mozilla-mobile/fenix/issues/14034).
```
Comment (rpl) [inline src/content/documentation/develop/developing-extensions-for-firefox-for-android.md:22]: Nit, are we sure that it is going to be a pref? maybe we should call it "a setting" (which will clarify then later once the implementation details for that plan are detailed enough), how that sounds?

File: src/content/documentation/develop/developing-extensions-for-firefox-for-android.md [hunk 2]
```diff
@@ -39,8 +41,43 @@ However, instructions are provided for the steps you need to take should you cho
...
+
+- Install or update to the latest version of [Firefox Nightly for Android](https://play.google.com/store/apps/details?id=org.mozilla.fenix).
+
+- Make sure you have adb installed ([Linux](https://dl.google.com/android/repository/platform-tools-latest-linux.zip), [Mac](https://dl.google.com/android/repository/platform-tools-latest-darwin.zip), [Windows](https://dl.google.com/android/repository/platform-tools-latest-windows.zip)) and in your PATH.
+
+- Connect your Android device via USB.
+
+- Make sure you have [Android's USB debugging](https://developer.android.com/studio/debug/dev-options) enabled. You can run `adb devices` to verify.
+
+- In the settings for Firefox Nightly, enable "Remote debugging via USB."
```
Comment (rpl) [inline src/content/documentation/develop/developing-extensions-for-firefox-for-android.md:59]: Nit, it happened at least once that the settings behind "Remote debugging via USB" was cleared across a Nightly update, it may be reasonable to rephrase this to something similar to the previous bullet point, something like `Make sure you have "Remove debugging via USB" enabled in the Firefox Nightly for Android`.

Maybe we should specify also that "Remote debugging via USB" is in the Firefox for Android Settings view?

File: src/content/documentation/develop/developing-extensions-for-firefox-for-android.md [hunk 3]
```diff
@@ -39,8 +41,43 @@ However, instructions are provided for the steps you need to take should you cho
...
+
+- In the unzipped directory of your extension, run `web-ext run -t firefox-android` and follow the instructions on screen to make sure you select the right device. Select `org.mozilla.fenix` as a target.
```
Comment (rpl) [inline src/content/documentation/develop/developing-extensions-for-firefox-for-android.md:61]: in the `web-ext run --help` output we call `target` the `-t firefox-android` option, and so it may be better to call this "apk" or "Firefox apk" (which will better match the name of the related cli option, `--firefox-apk`).

File: src/content/documentation/develop/developing-extensions-for-firefox-for-android.md [hunk 4]
```diff
@@ -39,8 +41,43 @@ However, instructions are provided for the steps you need to take should you cho
...
+<!-- Single Column Body Module -->
+
+{% capture content %}
+
 ## Set up your computer and Android emulator or device
 
+::: note alert
+This section only applies to Firefox for Android < 68 (Fennec)
+:::
+
```
Comment (rpl) [inline src/content/documentation/develop/developing-extensions-for-firefox-for-android.md:80]: The "on your development computer" part of this section does look actually not strictly related to Fennec (it is basically how to install adb), maybe we should split out (or just call out) the parts that are related to Fennec (which seems to be basically all the content in this subsection starting from "On you device or android emulator").

File: src/content/documentation/develop/developing-extensions-for-firefox-for-android.md [hunk 5]
```diff
@@ -83,16 +120,147 @@ If you are using a device:
...
+In your extension, ensure that you've included an application ID using the [browser_specific_settings](https://developer.mozilla.org/docs/Mozilla/Add-ons/WebExtensions/manifest.json/browser_specific_settings) key in the manifest.json:
+
+```json
+"browser_specific_settings": {
+  "gecko": {
+    "id": "borderify@example.com"
+  }
+}
```
Comment (rpl) [inline src/content/documentation/develop/developing-extensions-for-firefox-for-android.md:188]: This (an explicit id in the manifest) would still be necessary while using web-ext on Fenix

File: src/content/documentation/develop/developing-extensions-for-firefox-for-android.md [hunk 6]
```diff
@@ -83,16 +120,147 @@ If you are using a device:
...
+<!-- END: Single Column Body Module -->
+
+<!-- Single Column Body Module -->
+
+{% capture content %}
+
+## Check for Firefox for Android compatibility
+::: note alert
+This section only applies to Firefox for Android < 68 (Fennec)
+:::
```
Comment (rpl) [inline src/content/documentation/develop/developing-extensions-for-firefox-for-android.md:137]: It looks that most of this content was already part of this document and we are mainly adding the note alert about the section being related to Fennec, but it looks as added in this pull request.

My guess is that we didn't remove the existing content parts and that it is still below and everything after line 260 should be removed, would you mind to check that and update the PR to remove all the duplicated sections?

**Review by rpl — CHANGES_REQUESTED** [review]
@caitmuenster follows a round of review comments from my perspective.

File: src/content/documentation/develop/developing-extensions-for-firefox-for-android.md (hunk 6 above)
Comment (caitmuenster) [reply src/content/documentation/develop/developing-extensions-for-firefox-for-android.md:137]: Haha you are asking for the hard thing. :) I really struggled with tracking changes in the markdown file yesterday, so I started a [Google doc](https://docs.google.com/document/d/1jxytLShEa5zWcqmiRStDnJr0mMi5ouAtKA1JgHi9JUk/edit#) where we can stage the edits and then I will  update the PR.

File: src/content/documentation/develop/developing-extensions-for-firefox-for-android.md [hunk 7]
```diff
@@ -6,20 +6,22 @@ topic: Develop
...
 
-At the time of launch, only a limited number of [Recommended Extensions](https://support.mozilla.org/en-US/kb/recommended-extensions-program?utm_source=extensionworkshop.com&utm_medium=dev-article&utm_content=developing-extensions-for-firefox-for-android) will be supported. We would like to support more extensions and we are currently evaluating how to best do that. New announcements around mobile extension support will be posted on the [Add-ons Blog](https://blog.mozilla.org/addons/category/mobile?utm_source=extensionworkshop.com&utm_medium=dev-article&utm_content=developing-extensions-for-firefox-for-android).
+At the time of launch, only a limited number of [Recommended Extensions](https://support.mozilla.org/en-US/kb/recommended-extensions-program?utm_source=extensionworkshop.com&utm_medium=dev-article&utm_content=developing-extensions-for-firefox-for-android) will be supported. We would like to support more extensions and we are currently evaluating how to best do that.
```
[the remark this answers is not in the document]
Comment (caitmuenster) [reply src/content/documentation/develop/developing-extensions-for-firefox-for-android.md:20]: Oof, yes, good catch. Thank you!

File: src/content/documentation/develop/developing-extensions-for-firefox-for-android.md (hunk 6 above)
Comment (caitmuenster) [reply src/content/documentation/develop/developing-extensions-for-firefox-for-android.md:137]: Sorry @rpl, I commented earlier and then deleted the comment. I don't quite feel comfortable removing the Fennec info yet but maybe it makes more sense to move it to its own page.

File: src/content/documentation/develop/developing-extensions-for-firefox-for-android.md (hunk 2 above)
Comment (caitmuenster) [reply src/content/documentation/develop/developing-extensions-for-firefox-for-android.md:59]: I *think* this incoming edit addresses this nit? If not I might need more direction. 😅

File: src/content/documentation/develop/developing-extensions-for-firefox-for-android.md (hunk 3 above)
Comment (caitmuenster) [reply src/content/documentation/develop/developing-extensions-for-firefox-for-android.md:61]: Sorry, I don't understand what to change. :(

**Comment by caitmuenster** [comment]
Hey @rpl, thanks so much for your feedback. :) I made some bigger updates to this PR and I'm not sure how helpful the diff will be. Here's what has changed with the most recent commit: 

- There's now a separate page that is just about extension development for Fennec
- `/documentation/develop/developing-extensions-for-firefox-for-android/` is now focused on Fenix 
- I removed some information that looked like it didn't apply but (1) I probably didn't catch everything and (2) there is a chance that I took out some information that is still relevant. 

Let me know if it's easier to go over this on a call!

File: src/content/documentation/develop/developing-extensions-for-firefox-for-android.md [hunk 8]
```diff
@@ -83,18 +79,23 @@ If you are using a device:
...
 
+<!-- END: Single Column Body Module -->
+
 <!-- Single Column Body Module -->
 
 {% capture content %}
 
 ## Check for Firefox for Android compatibility
+::: note alert
+This section only applies to Firefox for Android < 68 (Fennec)
```
Comment (caitmuenster) [inline src/content/documentation/develop/developing-extensions-for-firefox-for-android.md:95]: @rpl fact check?

File: src/content/documentation/develop/developing-extensions-for-firefox-for-android.md [hunk 9]
```diff
@@ -210,6 +163,9 @@ Upload your xpi file to your website and make it accessible over HTTP. Browse to
 {% capture content %}
 
 ## Debug your extension
+::: note alert
+This section only applies to Firefox for Android < 68 (Fennec)
```
Comment (caitmuenster) [inline src/content/documentation/develop/developing-extensions-for-firefox-for-android.md:167]: @rpl fact check?

**Comment by caitmuenster** [comment]
Hey @Rob--W, I'm also adding you as a reviewer to this PR. :)

File: src/content/documentation/develop/developing-extensions-for-firefox-for-android-fennec.md [hunk 10]
```diff
@@ -0,0 +1,408 @@
...
+<!-- END: Page Hero Banner -->
+
+<!-- Content with Table of Contents Module -->
+
+{% capture content_with_toc %}
+
+## Set up your computer and Android emulator or device
+
+::: note alert
+This section only applies to Firefox for Android < 68 (Fennec)
```
Comment (Rob--W) [inline src/content/documentation/develop/developing-extensions-for-firefox-for-android-fennec.md:41]: Would it be more helpful to link to the latest do c from here? Or at least a pointer to the top of the doc that explains the deprecation+ correct doc location?

File: src/content/documentation/develop/developing-extensions-for-firefox-for-android-fennec.md [hunk 11]
```diff
@@ -0,0 +1,408 @@
...
+tags: [add-ons, beginner, guide, mobile, webextensions]
+contributors:
+  [caitmuenster, rebloor, juraj, mdnwebdocs-bot, ExE-Boss, Ding-Fan, andrewtruongmoz]
+last_updated_by: caitmuenster
+date: 2020-08-26
+---
+
+{% capture page_hero_banner_content %}
+
+# Developing extensions for Firefox for Android < version 68
```
Comment (Rob--W) [inline src/content/documentation/develop/developing-extensions-for-firefox-for-android-fennec.md:15]: "version ≤ 68" or "version 68 and earlier"

instead of "< version 68".

Similarly for everything else in this PR that references Fennec.

File: src/content/documentation/develop/developing-extensions-for-firefox-for-android.md [hunk 12]
```diff
@@ -143,59 +151,17 @@ In your extension, ensure that you've included an application ID using the [brow
...
-### Transfer your extension via a website
+```shell
+web-ext run -t firefox-android --adb-device XXX --firefox-apk org.mozilla.fenix
+```
 
-Upload your xpi file to your website and make it accessible over HTTP. Browse to the file and download it. Follow the installation instructions, which will be similar to those for an extension transferred using adb.
+<!-- END: Syntax Highlighting -->
 
 {% endcapture %}
 {% include modules/one-column.liquid
```
Comment (rpl) [inline src/content/documentation/develop/developing-extensions-for-firefox-for-android.md:167]: @caitmuenster as we agreed on Friday last week I double-checked the content of the section **"Debug you extension"**, a good chunk of the content can stay as it is, but there are some changes needed:

- add a note to explicitly mention that in Fenix the extensions are enabled lazily once at least one tab has been opened (mainly because it may surprise some user that none of the extensions code is loaded if they connect to the debugger before any tab has been actually opened)

- There is currently a Fenix-specific issue with using the devtools inspector on targeting the Fenix's browserActor popups ([Bug 1637616](https://bugzilla.mozilla.org/1637616)), it may be good to add a "warning" note at the start of the **Debug your extension** to point out this issue to the developers (which is unfortunately quite annoying, as a workaround a developer may temporarily change the extension to open the popup extension page into a tab to be able to inspect it using the developer tools) 

- we may change a bit the note at line 204-206 (`Unlike desktop Firefox, where content scripts are debugged in`), in my opinion we may just remove the `"Unlike desktop Firefox, ..."` and just say something like `"Content scripts can be reached and debugged when connected to the main process or to a tab target where a content script is running"` (the reason for this change is that in the meantime the devtools team has introduced multiprocess support to the Browser Toolbox and so the "reason why" this works is not the same as it was for Fennec and it is not actually even specific to mobile)

We may leave the rest of the content unmodified for now (there is currently no mention to connecting from about:debugging to a specific extension running on Fenix, which does actually also work).

File: src/content/documentation/develop/developing-extensions-for-firefox-for-android-fennec.md (hunk 10 above)
Comment (caitmuenster) [reply src/content/documentation/develop/developing-extensions-for-firefox-for-android-fennec.md:41]: Hey @Rob--W, I'm not sure what you mean by "do c." But I did see that a bunch of stuff got messed up when I copied over the old Fennec page, so I'm going to fix that!

Comment (Rob--W) [reply src/content/documentation/develop/developing-extensions-for-firefox-for-android-fennec.md:41]: Sorry, I typed that on mobile.

I meant "doc", to reference the most up to date documentation from here so that people who unknowingly arrived at the wrong page can quickly find the latest docs.

File: src/content/documentation/develop/developing-extensions-for-firefox-for-android.md (hunk 12 above)
Comment (caitmuenster) [reply src/content/documentation/develop/developing-extensions-for-firefox-for-android.md:167]: Thanks @rpl! I just pushed a new commit to make those changes. If you could re-review that would be great!

File: src/content/documentation/develop/developing-extensions-for-firefox-for-android-fennec.md (hunk 10 above)
Comment (caitmuenster) [reply src/content/documentation/develop/developing-extensions-for-firefox-for-android-fennec.md]: Yeah! I removed the warnings from the individual sections -- those were carry overs from when I tried to consolidate all information on one page -- and am just putting one warning at the top of the page with a link to the Fenix documentation. Can you take a look and let me know if that looks ok?

File: src/content/documentation/develop/developing-extensions-for-firefox-for-android.md [hunk 13]
```diff
@@ -39,62 +27,78 @@ However, instructions are provided for the steps you need to take should you cho
 
 {% capture content_with_toc %}
 
-## Set up your computer and Android emulator or device
+::: note alert
+In August 2020, Mozilla began rolling out a new, reimagined Firefox for Android experience (codenamed "Fenix"). The  browser for Android has been rebuilt from the ground up using [GeckoView](https://mozilla.github.io/geckoview/), Mozilla's mobile browsing engine.
 
-Complete some one-off setup tasks on your computer and Android device.
+At the time of launch, only a limited number of [Recommended Extensions](https://support.mozilla.org/kb/recommended-extensions-program?utm_source=extensionworkshop.com&utm_medium=dev-article&utm_content=developing-extensions-for-firefox-for-android) will be supported. We would like to support more extensions and we are currently evaluating how to best do that.
```
Comment (Rob--W) [inline src/content/documentation/develop/developing-extensions-for-firefox-for-android.md:33]: "will be" -> "were"

File: src/content/documentation/develop/differences-between-desktop-and-android-extensions.md [hunk 14]
```diff
@@ -28,9 +28,15 @@ date: 2020-08-04
...
 
-At the time of launch, only a limited number of [Recommended Extensions](https://support.mozilla.org/en-US/kb/recommended-extensions-program?utm_source=extensionworkshop.com&utm_medium=dev-article&utm_content=developing-extensions-for-firefox-for-android) will be supported. We would like to support more extensions and we are currently evaluating how to best do that. New announcements around mobile extension support will be posted on the [Add-ons Blog](https://blog.mozilla.org/addons/category/mobile?utm_source=extensionworkshop.com&utm_medium=dev-article&utm_content=developing-extensions-for-firefox-for-android).
+At the time of launch, only a limited number of [Recommended Extensions](https://support.mozilla.org/kb/recommended-extensions-program?utm_source=extensionworkshop.com&utm_medium=dev-article&utm_content=developing-extensions-for-firefox-for-android) will be supported. We would like to support more extensions and we are currently evaluating how to best do that.
```
Comment (Rob--W) [inline src/content/documentation/develop/differences-between-desktop-and-android-extensions.md:33]: "will be" -> "were"

File: src/content/documentation/develop/developing-extensions-for-firefox-for-android.md [hunk 15]
```diff
@@ -127,11 +131,15 @@ When setting `strict_min_version`, unless you’re targeting a specific version
...
 
 <!-- END: Single Column Body Module -->
+
 <!-- Single Column Body Module -->
 
 {% capture content %}
 
 ## Install and run your extension in Firefox for Android
+::: note alert
+Persistent loading is not yet supported on Firefox for Android 79+. These instructions are for temporary installation.
```
Comment (Rob--W) [inline src/content/documentation/develop/developing-extensions-for-firefox-for-android.md:141]: Add another warning:

> Unlike `web-ext` for desktop and Fennec, the add-on will be loaded in the main browser profile instead of a new temporary profile directory.

(maybe with some copy editing)

**Comment by rpl** [comment]
@caitmuenster lgtm, lets apply the changes described in the last round of comments from rob and update this branch to the current repo master.

As an additional side note (but not a big deal, we could consider it as an optional "nice to have"):

- in the new page for fenix the warning note and the introduction paragraph (`You'll approach ...`) are located after the TOC (the section with the "table of contents" links)
- in the page for fennec the warning note and the introduction paragraph (`You'll approach ...`) are right below the page title and before the TOC

it may be reasonable to use the same convention in both pages (after or before the TOC in both of them).

**Comment by caitmuenster** [comment]
Awesome, thank you both @Rob--W and @rpl! This latest commit should address all feedback so far. If you can r+ by EOD Wednesday, this can be included in the Sept 17 push.

**Review by Rob--W — APPROVED** [review]

**Review by rpl — APPROVED** [review]

File: src/content/documentation/develop/developing-extensions-for-firefox-for-android.md (hunk 15 above)
Comment (ghostwords) [reply src/content/documentation/develop/developing-extensions-for-firefox-for-android.md:141]: Loading in the main profile seems like a bug: https://github.com/mozilla/web-ext/issues/1950
